front/components/vaults/VaultCreateFolderModal.tsx

```diff
@@ -0,0 +1,128 @@
+import {
+  ExclamationCircleStrokeIcon,
+  Input,
+  Modal,
+  Page,
+  TextArea,
+} from "@dust-tt/sparkle";
+import type {
+  APIError,
+  DataSourceType,
+  VaultType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { isDataSourceNameValid } from "@dust-tt/types";
+import { useState } from "react";
+
+export default function VaultCreateFolderModal({
+  isOpen,
+  setOpen,
+  owner,
+  vault,
+  dataSources,
+}: {
+  isOpen: boolean;
+  setOpen: (isOpen: boolean) => void;
+  owner: WorkspaceType;
+  vault: VaultType;
+  dataSources: DataSourceType[];
+}) {
+  const [name, setName] = useState<string | null>(null);
+  const [nameError, setNameError] = useState<string | null>(null);
+
+  const [description, setDescription] = useState<string | null>(null);
+  const [descriptionError, setDescriptionError] = useState<string | null>(null);
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={() => {
+        setOpen(false);
+      }}
+      onSave={async () => {
+        if (!name) {
+          setNameError("Name is required.");
+          return;
+        }
+        const nameValidRes = isDataSourceNameValid(name);
+        if (nameValidRes.isErr()) {
+          setNameError(
+            "Name is invalid, must be multiple characters with no space."
+          );
+          return;
+        }
+        if (dataSources.find((ds) => ds.name === name)) {
+          setNameError("A data source with this name already exists.");
+          return;
+        }
+
+        if (!description || description.trim() === "") {
+          setDescriptionError("Description is required.");
+          return;
+        }
+
+        const res = await fetch(`/api/w/${owner.sId}/data_sources`, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            name,
+            description,
+            assistantDefaultSelected: false,
+            vaultId: vault.sId,
+          }),
+        });
+        if (res.ok) {
+          setOpen(false);
+        } else {
+          const err = (await res.json()) as { error: APIError };
+          setNameError(err.error.message);
```

Comment:
Can you use a `sendNotification` instead of setting an error on the name ? error could come from another reason. We can also send a notification when the folder is successfully created to be consistent ( like in CreateVaultModal )